test/hw_app_test.exs

```diff
@@ -0,0 +1,14 @@
+defmodule HWAppTest do
+  use ExUnit.Case
+  doctest HWApp
+
+  describe "#hello" do
+    test "when the name of somebody has been passed" do
+      assert HWApp.hello('Name') == "Hello Name!"
+    end
+
+    test "when the name has not been provided" do
+      assert HWApp.hello() == "Hello World!"
```

Comment:
This is already tested with a respective doctest, no need to duplicate the test here.